SwiftVoiceQuickstart/ViewController.swift

```diff
@@ -260,44 +260,41 @@ extension ViewController: UITextFieldDelegate {
 extension ViewController: PushKitEventDelegate {
     func credentialsUpdated(credentials: PKPushCredentials) {
         guard let accessToken = fetchAccessToken() else { return }
-        
-        let deviceToken = credentials.token.map { String(format: "%02x", $0) }.joined()
-        
-        let userDefaults = UserDefaults.standard
-        let cachedDeviceToken = userDefaults.string(forKey: kCachedDeviceToken) ?? deviceToken
-        
-        if cachedDeviceToken != deviceToken {
+
+        var cachedDeviceTokenData = UserDefaults.standard.data(forKey: kCachedDeviceTokenData)
+        if cachedDeviceTokenData != credentials.token {
```

Comment:
use swifty `if let` here?

also the check  `cachedDeviceTokenData != credentials.token` - does it do a deep comparision ?

Reply:
Yes, `!=` or `==` here are equal to `isEqual` in Obj-C.